octojobs/tests.py

```diff
@@ -0,0 +1,661 @@
+    new database session. It binds that session to the available engine
+    and returns a new session for every call of the dummy_request object.
+    """
+    SessionFactory = configuration.registry['dbsession_factory']
+    session = SessionFactory()
+    engine = session.bind
+
+    Base.metadata.drop_all(engine)
+    Base.metadata.create_all(engine)
+
+    def teardown():
+        session.transaction.rollback()
+
+    request.addfinalizer(teardown)
+    return session
+
+
+@pytest.fixture
+def dummy_request(db_session):
+    """It creates a fake HTTP Request and database session.
+
+    This is a function-level fixture, so every new request will have a
+    new database session.
+    """
+    return testing.DummyRequest(dbsession=db_session)
+
+
+@pytest.fixture
+def add_models(dummy_request):
+    """Add a bunch of model instances to the database.
+
+    Every test that includes this fixture will add jobs.
+    """
+    # dummy_request.dbsession.add_all(DUMMY_JOBS)
+
+    dummy_request.dbsession.add_all(DUMMY_JOBS)
+
+
+# ================ UNIT TESTS =======================
+
+def test_new_jobs_are_added(db_session):
+    """New Job objects get added to the database."""
+    db_session.add_all(DUMMY_JOBS)
+    query = db_session.query(Job).all()
+    assert len(query) == len(DUMMY_JOBS)
+
+
+def test_get_home_view_is_empty_dict(dummy_request):
+    """Assert empty dict is returned, from Get request."""
+    from octojobs.views.default import home_view
+    assert home_view(dummy_request) == {}
+
+
+def test_get_about_view_is_empty_dict(dummy_request):
+    """Assert empty dict is returned, from Get request."""
+    from octojobs.views.default import about_view
+    assert about_view(dummy_request) == {}
+
+
+def test_post_home_view_is_http_found(dummy_request):
+    """Assert is instance of HTTP found, from POST request on home."""
+    from octojobs.views.default import home_view
+
+    dummy_request.method = "POST"
+    dummy_request.POST["searchbar"] = "test"
+    dummy_request.POST["location"] = "seattle"
+
+    result = home_view(dummy_request)
+
+    assert isinstance(result, HTTPFound)
+
+
+def test_post_result_view_is_http_found(dummy_request):
+    """Assert is instance of HTTP found, from POST request on resultview."""
+    from octojobs.views.default import result_view
+
+    dummy_request.method = "POST"
+    dummy_request.POST["searchbar"] = "test"
+    dummy_request.POST["location"] = "seattle"
+
+    result = result_view(dummy_request)
+
+    assert isinstance(result, HTTPFound)
+
+
+def test_post_home_view_reroutes_with_url_query(dummy_request):
+    """Assert search terms are passed on url when a post request."""
+    from octojobs.views.default import home_view
+
+    dummy_request.method = "POST"
+    dummy_request.POST["searchbar"] = "test"
+    dummy_request.POST["location"] = "seattle"
+
+    result = home_view(dummy_request)
+
+    assert 'test' and 'seattle' in result.location
```

Comment:
```python
assert 'test' in result.location
assert 'seattle' in result.location
```

Otherwise Python evaluates as `test` being Truthy in addition to the result of `'seattle' in result.location`